### PR TITLE
Add `channel_settings` to documentation

### PR DESCRIPTION
**Pull request by travishathaway**
### Description

This moves `channel_settings` into the documentation.

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by travishathaway** [comment]
Add this setting to this page:

- https://docs.conda.io/projects/conda/en/stable/user-guide/configuration/settings.html

File: docs/source/user-guide/configuration/settings.rst [hunk 1]
```diff
@@ -64,6 +64,33 @@ repository, an administrator can set both :ref:`channel alias <channel-alias>` a
 
 .. _auto-update-conda:
 
+
+.. _channel-settings:
+
+``channel_settings``: Extra settings for individual channels
+------------------------------------------------------------
+
+With ``channel_settings``, it is possible to add extra configuration options
```
Comment (jezdez) [inline docs/source/user-guide/configuration/settings.rst:73]:
```suggestion

.. versionadded:: 23.3.0

With ``channel_settings``, it is possible to add extra configuration options
```

This should render a nice version notification, commonly used in Sphinx projects, https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded

**Review by jezdez — APPROVED** [review]

File: docs/source/user-guide/configuration/settings.rst (hunk 1 above)
Comment (travishathaway) [reply docs/source/user-guide/configuration/settings.rst:73]: Good idea!

Comment (jezdez) [reply docs/source/user-guide/configuration/settings.rst:73]: (we're sadly not very consistent with this)

Comment (travishathaway) [reply docs/source/user-guide/configuration/settings.rst:75]: This whole page needs actually some work 🙃. One step at a time.

**Comment by travishathaway** [comment]
@jezdez, ready for another look.

**Review by jezdez — APPROVED** [review]

**Pull request merged**